File: 4. Java/EX55.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX55
{
    public static void main(String[] args)
    {
        int m, n;
        double acuSom = 0; // Acumulador de soma
        
        Locale.setDefault(Locale.US); // Define ponto decimal
        // Entrada de dados
        try (Scanner ent = new Scanner(System.in))
        {
            // Solicita linhas e colunas
            System.out.print("# Qual a quantidade de linhas da matriz: ");
            m = ent.nextInt();
        
            System.out.print("# Qual a quantidade de colunas da matriz: ");
            n = ent.nextInt();
            
            double [][] mat = new double[m][n]; // Matriz
            double [] aux = new double[m];      // Soma das linhas
            
            // Preenche matriz e soma linhas
            for (int i = 0; i < m; i++)
            {
                System.out.printf("\n# Digite os elementos da %da linha: \n", i+1);
                
                for (int j = 0; j < n; j++)
                {
                    // Entrada dos elementos
                    System.out.print("# Elemento: ");
                    mat[i][j] = ent.nextDouble();
                    
                    acuSom += mat[i][j]; // Soma linha
                }
                
                aux[i] = acuSom; // Armazena soma
                acuSom = 0; // Reseta soma
            }
            
            // Exibe vetor com as somas
            System.out.println("\n# Vetor Gerado: ");
            for (int i = 0; i < m; i++)
            {
                System.out.println(aux[i]);
            }
        }
    }
}

File: 6. Python/EX55.py
# Função para obter entrada validada.
def protecao_entDados(txt, tipo):
    while True:
        try:
            a = tipo(input(txt))
            if a > 0:
                return a

        except Exception:
            print("\n# Entrada Invalida !")

# Função para popular matriz e calcular somas das linhas.
def popular_matriz(mat, aux):

    for i in range(len(mat)):

        acuSom = 0
        print(f"\n# Digite os elementos da {i+1}a. linha: ")

        for j in range(len(mat[0])):
            mat[i][j] = protecao_entDados("# Elemento: ", float)
            acuSom += mat[i][j]
        aux[i] = acuSom

    return aux

# Função para exibir vetor de somas.
def relatorio(aux):
    print("\n# Vetor Gerado: ")

    for i in range(len(aux)):
        print(aux[i])

# Função principal.
def main():
    m = protecao_entDados("# Qual a quantidade de linhas da matriz: ", int)
    n = protecao_entDados("# Qual a quantidade de colunas da matriz: ", int)

    aux = [0 for _ in range(m)]
    mat = [[0 for _ in range(n)] for _ in range(m)]

    relatorio(popular_matriz(mat, aux))

# Execução.
if __name__ == '__main__':
    main()